src/tests/end-to-end/tests/details-view/fast-pass-report.test.ts

```diff
@@ -0,0 +1,103 @@
+import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
+
+const reportDownloadUrl = getTestResourceUrl('fast-pass-report.html');
+
+describe('Details View -> FastPass -> Report', () => {
+    let browser: Browser;
+    let targetPage: TargetPage;
+    let detailsViewPage: DetailsViewPage;
+    let backgroundPage: BackgroundPage;
+    let reportPage: HtmlReportPage;
+    const reportSaveAsFilePath = 'src/tests/end-to-end/test-resources/fast-pass-report.html';
+    let reportDownload: Download;
+
+    beforeAll(async () => {
+        browser = await launchBrowser({
+            suppressFirstTimeDialog: true,
+            addExtraPermissionsToManifest: 'all-origins',
+        });
+        targetPage = await browser.newTargetPage();
+        await browser.newPopupPage(targetPage); // Required for the details view to register as having permissions/being open
+        backgroundPage = await browser.backgroundPage();
+        await backgroundPage.enableFeatureFlag(FeatureFlags.newTabStopsDetailsView);
+        detailsViewPage = await browser.newDetailsViewPage(targetPage);
+        await detailsViewPage.waitForSelector(detailsViewSelectors.automatedChecksResultSection);
+        await openTabStopsPage(detailsViewPage);
+        await addTabStopsFailure(detailsViewPage);
+        await detailsViewPage.waitForAndClickSelector(detailsViewSelectors.exportReportButton);
+        await detailsViewPage.waitForSelector(detailsViewSelectors.singleExportToHtmlButton);
+        reportDownload = await detailsViewPage.downloadExportReport(
+            detailsViewSelectors.singleExportToHtmlButton,
+            reportSaveAsFilePath,
+        );
+        reportPage = await browser.newHtmlReportPage(reportDownloadUrl);
+    });
+
+    afterAll(async () => {
+        // delete GUID file auto-generated by the download
+        await detailsViewPage.deleteDownloadedFile(reportDownload);
+        // delete renamed HTML used to open report for testing
+        await detailsViewPage.deleteFile(reportSaveAsFilePath);
+        await browser?.close();
+    });
+
+    it('should pass accessibility validation', async () => {
+        const results = await scanForAccessibilityIssues(reportPage, '*');
+        // should be expect(results).toHaveLength(0), but because of existing accessibility issue
+        // in the report, it's currently a snapshot. Once fixed we can remove the snapshot and check
+        // the length of the results array instead.
+        expect(results).toMatchSnapshot();
+    });
+
+    it('renders', async () => {
+        await reportPage.waitForSelector(fastPassReportSelectors.reportHeaderSection);
+    });
+});
+
+async function openTabStopsPage(detailsViewPage: DetailsViewPage): Promise<void> {
+    await detailsViewPage.waitForAndClickSelector(tabStopsSelectors.navDataAutomationId);
+}
+
+async function addTabStopsFailure(detailsViewPage: DetailsViewPage): Promise<void> {
```

Comment:
It's surprising to me that an earlier feature didn't add some of this infrastructure around e2e testing for the tab stops view. Do we know if leaving out e2e tests from the earlier tab stops UI feature was an intentional bit of tech debt? If not, we should probably note for a future retro that it would be good to circle back on how to avoid forgetting these in future features

Reply:
There was a fair bit done, it wasn't in a very reusable format. I've refactored it to use the same code in both places.